lib/checkexceptionsafety.h

```diff
@@ -135,6 +140,13 @@ class CPPCHECKLIB CheckExceptionSafety : public Check {
                     "Either use a try/catch around the function call, or add a exception specification for " + funcname + "() also.", CWE703, Certainty::inconclusive);
     }
 
+    /** Rethrow without currently handled exception */
+    void rethrowNoCurrentExceptionError(const Token *tok) {
+        reportError(tok, Severity::error, "rethrowNoCurrentException",
+                    "Calling `throw;` without currently handled exception(not from catch block) calls std​::​​terminate()",
```

Comment:
hmm.. I don't feel 100% satisfied about this error message. how about something like;

    rethrowing exception with 'throw;' outside a catch scope calls std::terminate().

Reply:
Yea, me either =). Thank you for better message